cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsClient.java

```diff
@@ -0,0 +1,96 @@
+import com.sequenceiq.cloudbreak.cloud.aws.view.AwsCredentialView;
+import com.sequenceiq.cloudbreak.cloud.event.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.event.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+
+@Component
+public class AwsClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsClient.class);
+
+    private static final int DEFAULT_SESSION_CREDENTIALS_DURATION = 3600;
+    private static final String DEFAULT_REGION_NAME = "US_WEST_1";
+
+    @Value("${cb.aws.external.id:" + CB_AWS_EXTERNAL_ID + "}")
+    private String externalId;
+
+    @Inject
+    private AwsPlatformParameters awsPlatformParameters;
+
+    public AuthenticatedContext createAuthenticatedContext(CloudContext cloudContext, CloudCredential cloudCredential) {
+        AuthenticatedContext authenticatedContext = new AuthenticatedContext(cloudContext, cloudCredential);
+        authenticatedContext.putParameter(AmazonEC2Client.class, createAccess(authenticatedContext.getCloudCredential()));
+        return authenticatedContext;
+    }
+
+    public AmazonEC2Client createAccess(CloudCredential credential) {
+        return createAccess(new AwsCredentialView(credential), DEFAULT_REGION_NAME);
+    }
+
+    public AmazonEC2Client createAccess(AwsCredentialView awsCredential, String regionName) {
+        BasicSessionCredentials basicSessionCredentials = retrieveCachedSessionCredentials(awsCredential);
+        AmazonEC2Client amazonEC2Client = new AmazonEC2Client(basicSessionCredentials);
+        amazonEC2Client.setRegion(RegionUtils.getRegion(awsPlatformParameters.regions().get(regionName)));
+        return amazonEC2Client;
+    }
+
+    public AmazonCloudFormationClient createCloudFormationClient(AwsCredentialView awsCredential, String regionName) {
+        BasicSessionCredentials basicSessionCredentials = retrieveCachedSessionCredentials(awsCredential);
+        AmazonCloudFormationClient amazonCloudFormationClient = new AmazonCloudFormationClient(basicSessionCredentials);
+        amazonCloudFormationClient.setRegion(RegionUtils.getRegion(awsPlatformParameters.regions().get(regionName)));
+        return amazonCloudFormationClient;
+    }
+
+    public AmazonAutoScalingClient createAutoScalingClient(AwsCredentialView awsCredential, String regionName) {
+        BasicSessionCredentials basicSessionCredentials = retrieveCachedSessionCredentials(awsCredential);
+        AmazonAutoScalingClient amazonAutoScalingClient = new AmazonAutoScalingClient(basicSessionCredentials);
+        amazonAutoScalingClient.setRegion(RegionUtils.getRegion(awsPlatformParameters.regions().get(regionName)));
+        return amazonAutoScalingClient;
+    }
+
+    public String getCbName(String groupName, Long number) {
+        return String.format("%s%s", groupName, number);
+    }
+
+    @Cacheable(AwsCachingConfig.TEMPORARY_AWS_CREDENTIAL_CACHE)
```

Comment:
How it is cached here? If it is cached based on AwsCredentialView the shouldn't we implement the hasCode + equals in AwsCredentialView.java? 

I think a generic/centralised service for credential caching would be better. That caching service could be used in OpenStack, Azure etc.